compendium/ServiceComponent/test/TestCompInst_RefOrder.cpp

```diff
@@ -111,9 +128,26 @@ namespace
         ComponentInstanceImpl<TestServiceImpl1,
                               std::tuple<>,
                               std::true_type,
-                              ServiceDependency2,
-                              ServiceDependency1>
-            compInstance(binders); // compile error
+                              std::shared_ptr<ServiceDependency2>,
+                              std::shared_ptr<ServiceDependency1>>
+            compInstance({ "bar", "foo" }, binders); // compile error
+    }
+
+    TEST(ComponentInstance, VerifyDependencyOrder_MultipleCardinality)
```

Comment:
this should be moved out into a separate compile time test EXE since it looks like the use of `set_tests_properties(usServiceComponentRefOrderTest PROPERTIES PASS_REGULAR_EXPRESSION )` only supports checking one compile error.

Reply:
Added a separate test file for this test case and added as a compile-only test.